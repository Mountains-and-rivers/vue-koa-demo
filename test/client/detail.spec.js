import { shallowMount } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import Detail from '@/components/Detail'

jest.mock('axios', () => ({
  get: jest.fn()
    // for test1
    .mockImplementationOnce(() => Promise.resolve({
      data: {
        todo: {
          todoDetail: 'this is todoDetail',
          todoTime: new Date().getTime(),
          todoState: 'done'
        }
      }
    }))
    // for test2
    .mockImplementationOnce(() => Promise.reject(new Error()))
    // for test3
    .mockImplementationOnce(() => Promise.resolve({
      data: {
        todo: {
          todoDetail: 'this is todoDetail',
          todoTime: new Date().getTime(),
          todoState: 'done'
        }
      }
    }))
}))

describe('Detail.vue', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })

  test('created: get detailTodo ok', async () => {
    const $route = { params: { todoId: '5c0361f44a240a3fa9e06928' } }
    const wrapper = shallowMount(Detail, {
      mocks: {
        $route
      }
    })
    await flushPromises()
    expect(wrapper.contains('.detail')).toBeTruthy()
    expect(wrapper.find('.detail-content').text()).toBe('this is todoDetail')
    expect(wrapper.contains('img')).toBe(true)
  })

  test('created: get detailTodo fail', async () => {
    const $route = { params: { todoId: '5c0361f44a240a3fa9e06928' } }
    const $router = { push: jest.fn() }
    const $message = { error: jest.fn() }
    const wrapper = shallowMount(Detail, {
      mocks: {
        $route,
        $router,
        $message
      }
    })
    await flushPromises()
    jest.runAllTimers()
    expect(wrapper.vm.$message.error).toBeCalled()
    expect(wrapper.vm.$router.push).toHaveBeenCalledTimes(1)
  })

  test('detail.vue snapshot test', () => {
    const $route = { params: { todoId: '5c0361f44a240a3fa9e06928' } }
    const wrapper = shallowMount(Detail, {
      mocks: {
        $route
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
