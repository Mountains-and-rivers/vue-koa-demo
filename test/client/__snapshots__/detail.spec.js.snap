// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail.vue detail.vue snapshot test 1`] = `
<div
  class="detail"
>
  <el-form-stub
    label-width="60px"
    model="[object Object]"
  >
    <el-form-item-stub
      label="时间"
    >
      <p
        class="detail-date"
      >
        
      </p>
    </el-form-item-stub>
     
    <el-form-item-stub
      label="内容"
    >
      <p
        class="detail-content"
      >
        
      </p>
    </el-form-item-stub>
     
    <!---->
     
    <i />
  </el-form-stub>
</div>
`;
